### PR TITLE
Make last announced routes nullable.

### PR DESCRIPTION
**Pull request by polina-c**
Contributes to https://github.com/dart-lang/leak_tracker/issues/218

### CODE REVIEW

**Comment by polina-c** [comment]
Getting test failure:

```
The following TestFailure was thrown running a test:
Expected: <0.9>
  Actual: <0.0>

When the exception was thrown, this was the stack:
#4      main.<anonymous closure>.<anonymous closure> (file:///Users/polinach/_/flutter_dev/packages/flutter/test/widgets/navigator_test.dart:3150:7)
```

File: packages/flutter/test/widgets/navigator_test.dart [hunk 1]
```diff
@@ -2341,11 +2341,11 @@ void main() {
 
     expect(previousOfFirst, isNull);
     expect(nextOfFirst, secondRoute);
-    expect(popNextOfFirst, isA<NotAnnounced>());
+    expect(popNextOfFirst, isNull);
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/flutter/test/widgets/navigator_test.dart]: Done. I had to change one line in the test to make it working. This means the behavior it changed.

So, https://github.com/flutter/flutter/pull/144050 is a cleaner way of this refactoring.

[the remark this answers is not in the document]
Comment (goderbauer) [reply packages/flutter/test/widgets/navigator_test.dart]: Sooo... is this changed behavior still correct? Or is this introducing a bug?

[the remark this answers is not in the document]
Comment (polina-c) [reply packages/flutter/test/widgets/navigator_test.dart]: It definitely introduced a bug. Thanks for guidance!

**Pull request closed without merging**